api/server.go

```diff
@@ -58,22 +63,40 @@ func (s *Server) Start() error {
 	if err != nil {
 		return errors.WithStack(err)
 	}
-	s.gsrv = grpc.NewServer()
+	opts, err := s.getTLSOpts()
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	s.gsrv = grpc.NewServer(opts...)
 	reflection.Register(s.gsrv)
 	service.RegisterPranaDBServiceServer(s.gsrv, s)
 	s.started = true
 	go s.startServer(list)
 	return nil
 }
 
+func (s *Server) getTLSOpts() ([]grpc.ServerOption, error) {
+	if len(s.tlsConfig.CACert.Contents) == 0 {
+		return []grpc.ServerOption{}, nil
```

Comment:
Why do we turn off TLS if no CA cert is provided? For TLS we should only need, at minimum a server cert and a server key?

Reply:
Fixed this. Now it checks the API TLS config is not empty to decide to turn TLS on/off. 

It requires CA cert to verify certificate signature.